core/src/test/java/com/datastax/oss/driver/internal/core/control/ControlConnectionTest.java

```diff
@@ -155,26 +154,23 @@ public void should_reconnect_if_channel_goes_down() throws Exception {
     CompletionStage<Void> initFuture = controlConnection.init(false, false, false);
     factoryHelper.waitForCall(node1);
 
-    waitForPendingAdminTasks();
-    assertThatStage(initFuture).isSuccess();
-    assertThat(controlConnection.channel()).isEqualTo(channel1);
-    verify(eventBus).fire(ChannelEvent.channelOpened(node1));
+    assertThatStage(initFuture)
+        .isSuccess(v -> assertThat(controlConnection.channel()).isEqualTo(channel1));
```

Comment:
Note that `assertThatStage` already waits (at most 2 seconds) if the future is not done yet. A lot of tests were not taking advantage of that.